C/test/yahdlc_test.cpp

```diff
@@ -3,7 +3,7 @@
 #include <boost/test/unit_test.hpp>
 #include "yahdlc.h"
```

Comment:
Understand. How about adding a yahdlc_set_state(state)/yahdlc_get_state() (most for unit testing)? Then the state can be static in yahdlc.c